CONTRIBUTING.md

```diff
@@ -1,292 +1,155 @@
 # Contributing to cuGraph
+cuGraph, and all of RAPIDS in general, is an open-source project where we encourage community involvement.  There are multiple ways to be involved and contribute to the cuGraph community, the top paths are listed below:
```

Comment:
The previous `contributing.md` was more consistent with other RAPIDS repositories. I wonder if this is something we want to preserve or not.

cudf and cuml:
https://github.com/rapidsai/cuml/blob/branch-0.14/CONTRIBUTING.md
https://github.com/rapidsai/cudf/blob/branch-0.14/CONTRIBUTING.md

Reply:
I'm not so worried about being consistent with the other projects, but I'll ask Keith/John about that

Reply:
I did move all the source build out, which was not consistent, and add the section on development environment.  I spoke with JohnZ and there is no issue with the documents drifting

Reply:
The cuDF and cuML files are also not that simalar